hash.h

```diff
@@ -23,9 +23,9 @@ uint32_t CalcHash(const std::string& name);
 struct Elf_Hash {
     uint32_t nbuckets{0};
     uint32_t nchains{0};
-    uint8_t tail[0];
+    uint8_t tail[1];
```

Comment:
Did you change the length of the array because zero-length arrays are allowed only in [GNU C as an extension](https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html)?

Reply:
Yes (I'm not sure, but I think https://en.wikipedia.org/wiki/Flexible_array_member is not ported to C++ yet)